### PR TITLE
Custom fields for Clients and Contacts

### PR DESCRIPTION
**Pull request by pierredup**
## Summary

Adds user-defined custom fields for Client and Contact records, replacing the legacy `ContactType` / `AdditionalContactDetail` system with a unified, polymorphic schema.

- New entities: `CustomField` (definition) and `CustomFieldValue` (value), polymorphic `(target, target_id)` discriminator scaling to additional entity types later
- 9 field types: text, textarea, number, date, email, URL, checkbox, single-select, multi-select — all serialized to a single TEXT column via `CustomFieldTypeResolver`
- Management page at `/settings/custom-fields` with two tabs (Client / Contact), drag-to-reorder via Sortable.js, modal create/edit, CSRF-guarded delete
- Field values rendered on Client view and per-Contact view via reusable `<twig:CustomFieldsList />` component with type-aware formatting
- Forms: `CustomFieldValueCollectionType` widget embedded in `ClientType` and `ContactType`; deferred-persistence via `postPersist`/`postFlush` listener so new records work despite Doctrine's `UlidGenerator` overwriting pre-set IDs
- API: `CustomField` exposed as full CRUD ApiResource with reorder operation; `customFields` object embedded on Client/Contact API payloads via normalizer/denormalizer/state-processor trio; OpenAPI schema decorator
- Migration `Version30100_1`: copies legacy `contact_types` / `contact_details` data into the new tables, then drops the old tables (irreversible by design); fresh-install seeder creates default `additional_email`, `phone`, `mobile` CONTACT fields
- Lifecycle: `preRemove` listener cleans up values when a Client/Contact is removed (preRemove instead of postRemove because Doctrine nullifies the ID before postRemove); `app:custom-fields:check-orphans` console command for operational recovery

Spec: `docs/superpowers/specs/2026-04-24-custom-fields-design.md`
Plan: `docs/superpowers/plans/2026-04-24-custom-fields.md`

## Notable design decisions

- **Polymorphic `target_id`**: scales cleanly to more entity types later. Trade-off — no DB-level FK on `target_id`; integrity enforced via the cleanup listener and an orphan-check command
- **Required-field semantics**: form-level only, not DB NOT NULL. Required fields apply only when rendered, so existing records aren't trapped by retroactive constraints
- **Hard-cascade delete with confirmation**: simpler than soft-archive; the index page shows usage counts and the delete modal warns before destroying values
- **MCP enablement**: deferred — API Platform 4 has `McpTool`/`McpResource` classes but the integration is more involved than the originally-planned `mcp: true` per-operation. Resources are exposed via standard `ApiResource` for now; explicit MCP exposure is a follow-up

## Test plan

- [x] `bin/ecs check` clean across all changes
- [x] `CustomFieldTypeResolverTest` (14 tests): all 9 types round-trip serialize/deserialize, edge cases (empty string, null, multi-select empty)
- [x] `CustomFieldTargetTest` / `CustomFieldTypeTest`: enum cases and helpers
- [x] `CustomFieldValueCollectionTypeTest`: form widget happy path, new-record persistence, ULID round-trip
- [x] `CustomFieldValueCleanupListenerTest`: values deleted when parent Client is removed
- [x] `CustomFieldsMigrationTest`: new tables exist, legacy tables gone
- [x] `CustomFieldApiTest` (3 tests): create/list, patch/delete, target filter
- [x] `DefaultDataTest`: install seeder creates 3 default CONTACT fields
- [x] `ContactTypeTest` / `ClientTypeTest`: form submission still works after embedding `customFields`
- [x] `ClientFormTest` snapshots updated for new Custom fields section
- [ ] Manual UI smoke: verify settings page renders, drag-reorder persists, create/edit/delete flows work, custom fields appear on client view/edit pages
- [ ] Functional/UI tests for settings page and client form integration are deferred (need browser-session setup that overlapped with pre-existing test failures)
- [ ] MCP smoke test deferred until MCP integration syntax is settled

Pre-existing test failures (Pdf mpdf issues, ClientTest/ContactTest API status NOT NULL constraint) confirmed identical on base branch — not regressions.

### CODE REVIEW

File: src/SettingsBundle/Action/CustomField/CreateAction.php [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+use Symfony\Component\HttpFoundation\Response;
+
+final class CreateAction extends AbstractController
+{
+    public function __construct(
+        private readonly FeatureGate $featureGate,
+    ) {
+    }
+
+    public function __invoke(): Response
```
[the remark this answers is not in the document]
Comment (pierredup) [reply src/SettingsBundle/Action/CustomField/CreateAction.php:30]: Un-used parameters should not be added for consistency
